source/elastic_problem.cc

```diff
@@ -0,0 +1,265 @@
+
+  // top face
+  VectorTools::interpolate_boundary_values (dof_handler,
+                                            2,
+                                            BoundaryValues<dim>(2),
+                                            boundary_values);
+
+  // bottom face
+  VectorTools::interpolate_boundary_values (dof_handler,
+                                            1,
+                                            BoundaryValues<dim>(1, timestep, dt, false, 2),
+                                            boundary_values);
+
+  // hull
+  VectorTools::interpolate_boundary_values (dof_handler,
+                                            0,
+                                            BoundaryValues<dim>(0, timestep, dt, true, 4),
+                                            boundary_values);
+
+  MatrixTools::apply_boundary_values (boundary_values,
+                                      system_matrix,
+                                      solution,
+                                      system_rhs);
+}
+
+template <int dim>
+void ElasticProblem<dim>::solve ()
+{
+  SolverControl           solver_control (1000, 1e-12);
+  SolverCG<>              cg (solver_control);
+  PreconditionSSOR<> preconditioner;
+  preconditioner.initialize(system_matrix, 1.2);
+  cg.solve (system_matrix, solution, system_rhs,
+            preconditioner);
+  hanging_node_constraints.distribute (solution);
+}
+
+template <int dim>
+void ElasticProblem<dim>::refine_grid ()
+{
+  Vector<float> estimated_error_per_cell (triangulation.n_active_cells());
+  KellyErrorEstimator<dim>::estimate (dof_handler,
+                                      QGauss<dim-1>(2),
+                                      typename FunctionMap<dim>::type(),
+                                      solution,
+                                      estimated_error_per_cell);
+  GridRefinement::refine_and_coarsen_fixed_number (triangulation,
+                                                   estimated_error_per_cell,
+                                                   0.3, 0.03);
+  triangulation.execute_coarsening_and_refinement ();
+}
+
+template <int dim>
+void ElasticProblem<dim>::run ()
+{
+    // create grid 
+    // cylinder from x = -3.5 to x = 1.6838 with radius = 1.3858
+    // height: 1.6838 + 3.5 = 5.1838 --> half height: 2.5919
+    //                      (triangulation, radius, half_length)
+    GridGenerator::cylinder (triangulation, 1.3858, 2.5919); 
+
+    // shift mesh in x direction such that the top face is at x = 1.6838
+    // 1.6838 - 2.5919 = -0.9081
+    Tensor<1, 3> shift_vec;
+    shift_vec[0] = -0.9081;
+    GridTools::shift(shift_vec, triangulation);
+
+    triangulation.set_all_manifold_ids_on_boundary(0,0);
+    static const CylindricalManifold<dim> cylM;
+    triangulation.set_manifold (0, cylM);
+
+    triangulation.refine_global (2);
+
+    // solve 
+    setup_system ();
+    assemble_system ();
+    solve ();
+
+    // tria = triangulation
```

Comment:
1. Will the `triangulation` be updated after the elastic problem is solved? If not, can I force that somehow?
2. How can I do something like in the comment in line 261? `tria` is a parameter (see line 66) and I want it to be the "deformed" `triangulation`.